res/enkf/es_update.py

```diff
@@ -1,44 +1,63 @@
-from cwrap import BaseCClass
-from res import ResPrototype
+from res import _lib
 
 
-class ESUpdate(BaseCClass):
-    TYPE_NAME = "es_update"
-    _smoother_update = ResPrototype(
-        "bool enkf_main_smoother_update(es_update, enkf_fs, enkf_fs)"
-    )
-
+class ESUpdate:
     def __init__(self, enkf_main):
-        assert isinstance(enkf_main, BaseCClass)
-
-        # enkf_main should be an EnKFMain, get the _RealEnKFMain object
-        real_enkf_main = enkf_main.parent()
-
-        super().__init__(
-            real_enkf_main.from_param(real_enkf_main).value,
-            parent=real_enkf_main,
-            is_reference=True,
-        )
+        self.ert = enkf_main
 
+    @property
     def _analysis_config(self):
-        return self.parent().analysisConfig()
+        return self.ert.analysisConfig()
 
     def hasModule(self, name):
         """
         Will check if we have analysis module @name.
         """
-        return self._analysis_config().hasModule(name)
+        return self._analysis_config.hasModule(name)
 
     def getModule(self, name):
         if self.hasModule(name):
-            self._analysis_config().getModule(name)
+            self._analysis_config.getModule(name)
         else:
             raise KeyError("No such module:%s " % name)
 
     def setGlobalStdScaling(self, weight):
-        self._analysis_config().setGlobalStdScaling(weight)
+        self._analysis_config.setGlobalStdScaling(weight)
 
     def smootherUpdate(self, run_context):
-        data_fs = run_context.get_sim_fs()
+        source_fs = run_context.get_sim_fs()
         target_fs = run_context.get_target_fs()
-        return self._smoother_update(data_fs, target_fs)
+
+        time_map = source_fs.getTimeMap()
+
+        last_step = time_map.getLastStep()
+        if last_step < 0:
+            last_step = self.ert.getModelConfig().get_last_history_restart()
+
+        step_list = [i for i in range(last_step + 1)]
```

Comment:
```suggestion
        step_list = list(range(last_step + 1))
```

I miss the old Python 2 days where `range` would get you a list and not a generator. :cry: